apps/prairielearn/src/batched-migrations/20241018171312_questions__share_publicly__backfill.ts

```diff
@@ -0,0 +1,28 @@
+import { makeBatchedMigration } from '@prairielearn/migrations';
+import { queryOneRowAsync, queryAsync } from '@prairielearn/postgres';
+
+export default makeBatchedMigration({
+  async getParameters() {
+    const result = await queryOneRowAsync('SELECT MAX(id) as max from questions;', {});
+    return {
+      min: 1n,
+      max: result.rows[0].max,
+      batchSize: 1000,
+    };
+  },
+
+  async execute(start: bigint, end: bigint): Promise<void> {
+    await queryAsync(
+      `
+      UPDATE questions AS q
+      SET
+        share_publicly = q.shared_publicly
+      FROM
+        questions AS q
+      WHERE
+        v.id >= $start AND
+        v.id <= $end`,
```

Comment:
I'd probably move this to a separate `.sql` file. I know I've been inconsistent with that in other batched migrations, but I think we should standardize on that like we have for the rest of the codebase (with perhaps the exception of one-line queries like the `SELECT MAX(...)` above).

Reply:
done